session.go

```diff
@@ -33,6 +33,15 @@ type Session struct {
 	client           bool
 }
 
+// Use this defined type so we can share context between remotedialer and its clients
```

Comment:
https://pkg.go.dev/context#WithValue

Specifically says not to use a string as the key type, while there's a low possibility of collision here, the pattern is generally to use a `struct{}` as the key.

For example...
 * https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/containerd/log/context.go#L60
 * https://github.com/kubernetes/kubernetes/blob/master/vendor/k8s.io/utils/trace/trace.go#L303